router/core/operation_processor.go

```diff
@@ -308,6 +327,13 @@ func (o *OperationKit) unmarshalOperation() error {
 			}
 		}
 		if o.parsedOperation.GraphQLRequestExtensions.PersistedQuery != nil {
+			if !o.parsedOperation.GraphQLRequestExtensions.PersistedQuery.isValidHash() {
+				return &httpGraphqlError{
+					message:    "persistedQuery.sha256Hash must be a valid sha256 hash",
```

Comment:
I would expect this error message to surface somewhere in a test. Currently it looks like it's swallowed or not yet tested.

Reply:
Messages from all the errors returned from the `unmarchalOperation` are rewritten by this calling block:

```
	} else if req.Method == http.MethodPost {
		if err := operationKit.UnmarshalOperationFromBody(httpOperation.body); err != nil {
			return &httpGraphqlError{
				message:    "error parsing request body",
				statusCode: http.StatusBadRequest,
			}
		}
```
`UnmarshalOperationFromBody` returns errors from `unmarchalOperation`
It means that those internal errors act like a kind of documentation right now.

Reply:
That is the reason why we see only `"error parsing request body"` errors in the tests.

Reply:
@jensneuse 
I have changed how we handle errors returned by `operationKit.UnmarshalOperation*` methods. 
Now user see the error returned by the "UnmarshalOperationFromBody". 
I added couple of tested for "UnmarshalOperationFromURL".
Please review.